pennylane_lightning_gpu/lightning_gpu.py

```diff
@@ -229,7 +229,7 @@ def state(self):
             """Copy the state vector data from the device to the host. A state vector Numpy array is explicitly allocated on the host to store and return the data."""
             state = np.zeros(1 << self.num_wires, dtype=self.C_DTYPE)
             state = self._asarray(state, dtype=self.C_DTYPE)
-            self.syncH2D(state)
+            self.syncD2H(state)
```

Comment:
Can we add a quick test to verify an arbitrary state can be copied back?
Even the dosctring could be a valid test.